segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +219,142 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
+        region_centers: a dictionary mapping each cell marker to its respective center (also a tuple format)
+    """
+
+    # test that the radius specified is within the boundaries
+    # to prevent any overflow, we ask that the radius be no larger than
+    # half of any dimension passed
+    if radius > size_img[0] / 2 or radius > size_img[1] / 2:
+        raise ValueError("Radius specified is larger than one of the image dimensions")
+
+    # generate the two cells at the top left of the image
+    center_1 = (radius, radius)
+    center_2 = (radius, radius * 3 + 1)
+
+    # draw the coordnates covered for the two cell
+    x_coords_cell_1, y_coords_cell_1 = circle(center_1[0], center_1[1], radius + 1)
+    x_coords_cell_2, y_coords_cell_2 = circle(center_2[0], center_2[1], radius + 1)
+
+    # now store the regions generated by skimage.draw.circle into a dict with the key
+    # identifying the marker
+    cell_regions = {1: (x_coords_cell_1, y_coords_cell_1),
+                    2: (x_coords_cell_2, y_coords_cell_2)}
+
+    # now store the centers of each marker in a cell
+    # we're not storing the radius because that parameter is passed into generate_test_channel_data
+    # which calls this function, meaning it is already known
+    cell_centers = {1: center_1, 2: center_2}
+
+    return cell_regions, cell_centers
+
+
+def generate_two_cell_test_channel_data(size_img=(1024, 1024), radius=10):
+    """
+    This function generates test channel data assuming we're just generating two cells.
+
+    Args:
+        size_img: the dimensions of the image we wish to generate
+        radius: the radius of the cells we wish to generate
+
+    Returns:
+        sample_channel: a m x n x p array where m is the number of channels and (n x p)
+            is the same as size_img. We'll have 2 channels: nuclear and membrane.
+    """
+
+    cell_regions, cell_centers = generate_two_cell_test_regions(size_img=size_img, radius=radius)
+
+    # create the NumPy array we'll store the channel data in
+    sample_channel_data = np.zeros((2, size_img[0], size_img[1]))
+
+    # generate the nuclear- and membrane-level channel signal for each cell
+    for i in range(2):
+        sample_channel_data[0, :, :] += generate_two_cell_test_signal_data(size_img=size_img, radius=radius,
+                                                                           expression='nuclear', pattern='nuclear',
+                                                                           cell_region=cell_regions[i + 1], cell_center=cell_centers[i + 1])
+
+        sample_channel_data[1, :, :] += generate_two_cell_test_signal_data(size_img=size_img, radius=radius,
+                                                                           expression='membrane', pattern='membrane',
+                                                                           cell_region=cell_regions[i + 1], cell_center=cell_centers[i + 1])
```

Comment:
I think you can do something like
```
for cr, cc in zip(cell_regions, cell_centers):
```
for a teenytiny speed up

Reply:
Ah yes, thanks for that suggestion.